lib/features/context-pad/ContextPad.js

```diff
@@ -106,10 +108,30 @@ ContextPad.prototype._init = function() {
 /**
  * Register a provider with the context pad
  *
+ * @param  {Number} [priority=1000]
  * @param  {ContextPadProvider} provider
+ *
+ * @example
+ * contextPad.registerProvider(800, {
+  *   getContextPadEntries: function(element) {
+  *     return function(entries) {
+  *       return Object.assign(entries, {
+  *         'entry-1': {
+  *           label: 'My Entry',
+  *           action: function() { alert("I have been clicked!"); }
+  *         }
+  *       });
+  *     }
+  *   }
+  * });
  */
-ContextPad.prototype.registerProvider = function(provider) {
-  this._providers.push(provider);
+ContextPad.prototype.registerProvider = function(priority, provider) {
+  if (!provider) {
+    provider = priority;
+    priority = DEFAULT_PRIORITY;
+  }
+
+  this._insertProviderByPriority(provider, priority);
```

Comment:
I'd like to use the eventBus here, as it handles priorities already. Will prepare a proposal regarding that.

Reply:
Makes sense, I should've thought about this.